Packages/com.chisel.editor/Chisel/Editor/Editor/Scene/CSGSceneBottomGUI.cs

```diff
@@ -13,670 +13,98 @@ namespace Chisel.Editors
-		}
-		#endregion
-
-		static void DrawLeftButtons(Rect position, SceneViewBarState state)
-		{
-			directionForward = true;
-
-			position.x--;
-
-			ViewOptionsGUI(ref position);
-			if (!state.hideLocking) AxisLockingGUI(ref position, state.showLockLabel);
-			if (!state.hideMove) MoveSnappingGUI(ref position, state.largeLabels, state.showUnits, state.showIncDec, state.showAxiLabels, state.forceOneAxis);
-			if (!state.hideRotate) RotateSnappingGUI(ref position, state.largeLabels, state.showUnits, state.showIncDec);
-			if (!state.hideScale) ScaleSnappingGUI(ref position, state.largeLabels, state.showUnits, state.showIncDec);
-		}
-
-		static float CalcLeftButtonsWidth(SceneViewBarState state)
-		{
-			return ViewOptionsWidth() +
-				   (!state.hideLocking ? AxisLockingWidth(state.showLockLabel) : 0) +
-				   (!state.hideMove ? MoveSnappingWidth(state.largeLabels, state.showUnits, state.showIncDec, state.showAxiLabels, state.forceOneAxis) : 0) +
-				   (!state.hideRotate ? RotateSnappingWidth(state.largeLabels, state.showUnits, state.showIncDec) : 0) +
-				   (!state.hideScale ? ScaleSnappingWidth(state.largeLabels, state.showUnits, state.showIncDec) : 0);
-		}
-
-		static float CalcRightButtonsWidth()
-		{
-			return kButtonSpace +
-				   StyleWidth(EditorStyles.toolbarButton, rebuildContent);
-		}
-
-		static void DrawRightButtons(Rect position)
-		{
-			directionForward = false;
-
-			position.x = position.width;
-
-			Space(ref position, kButtonSpace);
-
-			//
-			// Rebuild all
-			//
-			if (Button(ref position, rebuildContent))
-			{
+			if (GUILayout.Button("Rebuild", buttonStyle)) { 
 				Rebuild();
 			}
-		}
 
-		static float CalcBottomBarWidth(SceneViewBarState state)
-		{
-			return CalcLeftButtonsWidth(state) +
-				   CalcRightButtonsWidth();
-		}
-
-		class SceneViewBarState
-		{
-			public bool     largeLabels     = true;
-			public bool     showLockLabel   = true;
-			public bool     showUnits       = true;
-			public bool     showIncDec      = true;
-			public bool     showAxiLabels   = true;
-			public bool     forceOneAxis    = false;
-			public bool     hideScale       = false;
-			public bool     hideRotate      = false;
-			public bool     hideLocking     = false;
-			public bool     hideMove        = false;
-			public float    prevBarWidth    = 0;
-
-			public void Reset(float width)
-			{
-				largeLabels = true;
-				showLockLabel = true;
-				showAxiLabels = true;
-				showUnits = true;
+			GUILayout.FlexibleSpace();
 
-				showIncDec = true;
-				forceOneAxis = false;
-				hideScale = false;
-				hideRotate = false;
-				hideLocking = false;
-				hideMove = false;
+			CSGEditorSettings.ShowGrid = GUILayout.Toggle(CSGEditorSettings.ShowGrid, "Show Grid", toggleStyle);
 
-				prevBarWidth = width;
+			CSGEditorSettings.MoveSnapX = CSGEditorSettings.MoveSnapY = CSGEditorSettings.MoveSnapZ = EditorGUILayout.FloatField(CSGEditorSettings.MoveSnapX, floatWidthLayout);
```

Comment:
Maybe add a MoveUniformSnap (or something) property to CSGEditorSettings that sets x,y & z